### PR TITLE
feat: add ResourceExhausted as separate error metric in GrpcServerMon…

### PR DESCRIPTION
**Pull request by Tasssadar**
…itoringInterceptor

### CODE REVIEW

File: grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java [hunk 1]
```diff
@@ -37,19 +37,17 @@ public void close(final Status status, final Metadata trailers) {
                 final Duration duration = Duration.between(start, clock.instant());
                 currentCalls.decrementAndGet();
 
-                if (ErrorCategory.fatal.contains(status.getCode())) {
-                    cache.getTimer(method, "FatalServerFailures")
-                            .update(duration);
+                String metricName = "ServerFailures";
+                if (status.isOk()) {
+                    metricName = "Successes";
+                } else if(status.getCode() == Status.Code.RESOURCE_EXHAUSTED) {
```
Comment (augi) [inline grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java:43]:
```suggestion
                } else if (status.getCode() == Status.Code.RESOURCE_EXHAUSTED) {
```

File: grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java [hunk 2]
```diff
@@ -37,19 +37,17 @@ public void close(final Status status, final Metadata trailers) {
...
+                    metricName = "ResourceExhausteds";
```
Comment (augi) [inline grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java:44]:
```suggestion
                    metricName = "ResourceExhausted";
```

Comment (augi) [reply grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java:44]: Maybe not 🤔

**Review by augi — APPROVED** [review]

File: grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java (hunk 2 above)
Comment (Tasssadar) [reply grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java:44]: Yeah all the others are plurals :D

**Pull request merged**